### PR TITLE
Fix TokenCreate Specs

### LINKED ISSUE #8757: Fix TokenCreateSpec.validateNewTokenAssociations
I debugged it for a while but I couldn't figure it out.

```
tokenCreate(notToBeToken)
        .treasury(tbd)
        .hasKnownStatus(INVALID_TREASURY_ACCOUNT_FOR_TOKEN)
        .via(failedCreationTxn))
```
The above tokenCreate is failing because we expected INVALID_TREASURY_ACCOUNT_FOR_TOKEN but it was OK.
When I change INVALID_TREASURY_ACCOUNT_FOR_TOKEN with OK - it starts throwing PLATFORM_NOT_ACTIVE.

### LINKED ISSUE #8758: Fix TokenCreateSpec.autoRenewValidationWorks
Not sure if it's an issue with the test setup or with the `autoRenewPeriod` implementation but for some reason for autoRenewPeriod we always get 7776000(the default value).

### LINKED ISSUE #8811: Fix TokenCreateSpecs.onlyValidCustomFeeScheduleCanBeCreated test
The part of the test that is failing is this one:
`.withCustom(fixedHtsFee(htsAmount, SENTINEL_VALUE, htsCollector))`
The business logic is that if we create a token with a fixed custom fee and we have a SENTINEL value for the denomination we should link with the token:
`.hasCustom(fixedHtsFeeInSchedule(htsAmount, token, htsCollector))`.

It seems that we have the logic in place: `CustomFeeValidator.validateFixedFeeForCreation`. Comparing it with mono(`FixedFeeSpec.validateAndFinalizeWith`) the logic seems the same. Maybe something down the road is breaking up but I can't figure it out.

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #8811 
Fixes #8758 
Fixes #8757 
Fixes https://github.com/hashgraph/hedera-services/issues/9175

Fixes a few TokenCreateSpecs

### CODE REVIEW

**Review by vtronkov — APPROVED** [review]
LGTM

**Review by tinker-michaelj — DISMISSED** [review]
Ah great catch, thanks @Neeharika-Sompalli !

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java [hunk 1]
```diff
@@ -162,24 +170,26 @@ public void handle(@NonNull final HandleContext context) {
...
      * is set to sentinel value, to use denomination as newly created token.
-     * @param newToken newly created token
-     * @param accountStore account store
-     * @param tokenRelStore token relation store
+     *
+     * @param newToken                        newly created token
+     * @param accountStore                    account store
+     * @param tokenRelStore                   token relation store
      * @param requireCollectorAutoAssociation set of custom fees whose token denomination is set to sentinel value
+     * @param recordBuilder
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java:184]: Nit: missing javadoc for this param

**Review by mhess-swl — DISMISSED** [review]

**Review by vtronkov — APPROVED** [review]

**Review by vtronkov — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
Nice catch on that topic backward validation 🥂 , LGTM!

**Pull request merged**